src/main/java/mServer/crawler/sender/ard/ArdVideoDeserializer.java

```diff
@@ -21,74 +23,122 @@
-                    streamArray.forEach(streamItem -> {
-                        deserializeMediaStreamArrayItem(streamItem.getAsJsonObject(), urlMap);
-                    });
-                });
+                if (mediaArray.size() > 0) {
+                  JsonElement mediaArrayItem = mediaArray.get(mediaArray.size() - 1);
+                  JsonObject item = mediaArrayItem.getAsJsonObject();
+                  JsonArray streamArray = item.get(JSON_OBJECT_MEDIASTREAMARRAY).getAsJsonArray();
+                  deserializeMediaStreamArray(streamArray, urlMap);
+                }
                 
                 // URLs setzen
                 urlMap.forEach((key, value) -> {
-                    dto.addVideo(key, value.url);
+                    dto.addVideo(key, value);
                 });
             }
         }
         
         return dto;
     }    
     
-    private void deserializeMediaStreamArrayItem(JsonObject stream, Map<Qualities, UrlWidthHeightDTO> urlMap) {
+    private void deserializeMediaStreamArray(JsonArray streamArray, Map<Qualities, String> urlMap) {
+      List<QualityUrlWidthHeightDTO> qualities = new ArrayList<>();
+      
+      streamArray.forEach(streamItem -> {
+        qualities.add(deserializeUrlInfos(streamItem.getAsJsonObject()));
+      });
+      
+      convertQualities(qualities, urlMap);
+    }
+    
+    private void convertQualities(List<QualityUrlWidthHeightDTO> qualities, Map<Qualities, String> urlMap) {
+      // bei der besten Qualität anfangen
+      qualities.sort((a, b) -> b.quality.compareTo(a.quality));
+      
+      for (int i = 0; i < qualities.size(); i++) {
+        QualityUrlWidthHeightDTO qualityDto = qualities.get(i);
+        if (qualityDto.urls.isEmpty()) {
+          continue;
+        }
         
-        String qualityValue = stream.get(JSON_ELEMENT_QUALITY).getAsString();
-
-        UrlWidthHeightDTO urlInfos = deserializeUrlInfos(stream);
-
-        Qualities quality = convertQuality(qualityValue, urlInfos);
-        // nur relevante Qualitäten berücksichtigen und nur http-Links (manchmal sind "mp3:"-Links enthalten)
-        if (quality != null && urlInfos.url.startsWith(TEXT_START_HTTP)) {
-            if (urlMap.containsKey(quality)) {
-                UrlWidthHeightDTO actualUrlInfos = urlMap.get(quality);
-                
-                // prüfen, ob die Auflösung besser ist, als die bisher für die Qualität hinterlegte
-                if (urlInfos.width > 0 && urlInfos.width > actualUrlInfos.width) {
-                   urlMap.put(quality, urlInfos);
-                } else if (quality == Qualities.NORMAL) {
-                    // bei normaler Auflösung ist aus irgendeinem Grund der letzte
-                    // Eintrag immer der beste!
-                    urlMap.put(quality, urlInfos);
-                }
-            } else {
-                urlMap.put(quality, urlInfos);
+        switch(qualityDto.quality){
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Add a default case to this switch. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ASwitchLastCaseIsDefaultCheck)